src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -111,6 +102,43 @@ object Content {
       None
     }
   }
+
+  /**
+   * Apply `func` to each file in directory recursively.
+   *
+   * @param prefixPath Root prefix
+   * @param directory Root directory
+   * @param func Function which would apply to current prefix and file
+   * @tparam T
+   * @return Result list of applying function to all files
+   */
+  private[hyperspace] def recFilesApply[T](
```

Comment:
Could you move this function back to `case class Content`? as this is a private function that should be only used in `case class Content`.

Reply:
Hi, sorry for delay. I consider it as just static function, so it could live separately from class. It also simplify testing

Reply:
I'm fine with this being a generic utility function, but:

1) The name `recFilesApply` seems unusual. Maybe `applyToFilesRecursively`?
2) The meaning of "prefix" here seems unclear. What is the "root prefix" and what is the "current prefix"? Without looking at the code, it is hard to catch the meaning of the parameters.

Reply:
Thanks @clee704. Valid point, I have inherited `prefixPath` from original function. I have changed it to just `path` and added a short description why we need it